public/app/plugins/panel/logs/LogsPanel.test.tsx

```diff
@@ -252,33 +253,118 @@ describe('LogsPanel', () => {
       });
     });
   });
+
+  describe('Performance regressions', () => {
+    const series = [
+      createDataFrame({
+        refId: 'A',
+        fields: [
+          {
+            name: 'time',
+            type: FieldType.time,
+            values: ['2019-04-26T09:28:11.352440161Z'],
+          },
+          {
+            name: 'message',
+            type: FieldType.string,
+            values: ['logline text'],
+            labels: {
+              app: 'common_app',
+              job: 'common_job',
+            },
+          },
+        ],
+      }),
+    ];
+
+    beforeEach(() => {
+      jest.spyOn(styles, 'getLogRowStyles');
+      jest.mocked(styles.getLogRowStyles).mockClear();
+    });
+
+    it('does not rerender without changes', async () => {
```

Comment:
Maybe it's worth adding a comment that we use `getLogRowStyles` as a way to count the re-renders for sake of a better way?

Reply:
Definitely.